proto/regen/curation/v1/tx.proto

```diff
@@ -0,0 +1,132 @@
+    rpc EditTagSetMetadata(MsgEditTagSetMetadata) returns (MsgEditTagSetMetadataReponse) {}
+
+    // ChangeTagSetMaintainer changes the maintainer of a tag set.
+    rpc ChangeTagSetMaintainer(MsgChangeTagSetMaintainer) returns (MsgChangeTagSetMaintainerResponse) {}
+
+    // EditTagMetadata edits a tag's metadata.
+    rpc EditTagMetadata(MsgEditTagMetadata) returns (MsgEditTagMetadataReponse) {}
+}
+
+// MsgCreateTagSet is the Msg/CreateTagSet request type.
+message MsgCreateTagSet {
+    // maintainer is the address of the tag set maintainer.
+    string maintainer = 1;
+
+    // name is an optional human readable name of the tag set.
+    string name = 2;
+
+    // description is an optional human readable description of the tag set.
+    string description = 3;
+
+    // uri is an optional URI/URL describing the tag set.
+    string uri = 4;
+}
+
+// MsgCreateTagSetResponse is the Msg/CreateTagSetResponse response type.
+message MsgCreateTagSetResponse {
+    // tag_set_id is the auto-generated ID of the tag set.
+    string tag_set_id = 1;
+}
+
+// MsgCreateTag is the Msg/CreateTag request type.
+message MsgCreateTag {
+    // maintainer is the address of the tag set maintainer.
+    string maintainer = 1;
+
+    // tag_set_id is the ID of the tag set within which this tag should be created.
+    string tag_set_id = 2;
+
+    // name is an optional human readable name of the tag .
+    string name = 3;
+
+    // description is an optional human readable description of the tag.
+    string description = 4;
+
+    // uri is an optional URI/URL describing the tag.
+    string uri = 5;
+}
+
+// MsgCreateTagResponse is the MsgCreateTagResponse response type.
+message MsgCreateTagResponse {
+    // tag_id is the auto-generated ID of the tag. It is always prefixed by the tag set ID for the tag
+    // set within which it was created and a `/` character.
+    string tag_id = 1;
+}
+
+// MsgTagAsset is the Msg/TagAsset request type.
+message MsgTagAsset {
+    // maintainer is the address of the tag's maintainer (this is always the same as the tag set's maintainer).
+    string maintainer = 1;
+
+    // tag_id is the ID of the tag.
+    string tag_id = 2;
+
+    // denom is the denom of the asset.
+    string denom = 3;
+}
+
+// MsgTagAssetResponse is the Msg/TagAssetResponse response type.
+message MsgTagAssetResponse { }
+
+message MsgTagAssetClass {
+    // maintainer is the address of the tag's maintainer (this is always the same as the tag set's maintainer).
+    string maintainer = 1;
+
+    // tag_id is the ID of the tag.
+    string tag_id = 2;
+
+    // denom_prefix is the prefix of denom's in this asset class. This could correspond to a credit class prefix which
+    // is always the prefix of all credit batches issued within this class. This provides a way to tag whole classes of
+    // credits or NFTs.
+    string denom_prefix = 3;
+}
+
+message MsgTagAssetClassResponse { }
+
+message MsgEditTagSetMetadata {
```

Comment:
what if we only want to edit one field (eg name)? Should we just have the rest of the fields set with their current values?